src/NUnit-Executor/NUnitExecutor.cs

```diff
@@ -1,61 +1,169 @@
+        /// Initializes a new instance of the <see cref="NUnitExecutorPlugin"/> class.
         /// </summary>
-        /// <param name="options">Options from the command line.</param>
-        /// <param name="computer">Target Computer where the tests will be run.</param>
-        /// <param name="testQueue">Test queue.</param>
-        /// <param name="nunitPath">Nunit path.</param>
-        public NUnitExecutorPlugin(
-            IReadOnlyOptions options,
-            IComputer computer,
-            TestQueue testQueue,
-            string nunitPath)
+        /// <param name="xmlConfig">NTestController.xml file path.</param>
+        public NUnitExecutorPlugin(string xmlConfig)
         {
-            _options = options;
-            _computer = computer;
-            _testQueue = testQueue;
-            _nunitPath = nunitPath;
+            ThrowIf.StringIsNullOrWhiteSpace(xmlConfig, nameof(xmlConfig));
+
+            _xmlConfig = xmlConfig;
         }
 
-        #region Inherited from IPlugin
+        #endregion Constructors
+
+        #region Inherited from IExecutorPlugin
+
+        public string Name { get; private set; }
+
+        public PluginType PluginType
+        {
+            get { return PluginType.TestExecutor; }
+        }
 
-        public string Name { get; }
-        public PluginType PluginType { get; }
+        public IReadOnlyOptions Options { get; set; }
+        public IComputer Computer { get; set; }
+        public TestQueue TestQueue { get; set; }
 
         /// <seealso cref="IPlugin.Execute()"/>
         public bool Execute()
         {
-            var test = _testQueue.DequeueTestToRun();
+            Log.WriteInfo("===== Starting tests on Computer: {0}", Computer.Hostname);
+
+            var test = TestQueue.DequeueTestToRun() as NUnitTest;
 
             while (test != null)
             {
                 RunTest(test);
-                test = _testQueue.DequeueTestToRun();
+                test = TestQueue.DequeueTestToRun() as NUnitTest;
             }
 
+            Log.WriteInfo("===== Finished all tests on Computer: {0}", Computer.Hostname);
+
             return true;
         }
 
-        #endregion Inherited from IPlugin
+        /// <seealso cref="IExecutorPlugin.ClonePlugin()"/>
+        public IExecutorPlugin ClonePlugin()
+        {
+            var newPlugin = new NUnitExecutorPlugin(_xmlConfig);
+            newPlugin.Options = Options;
+
+            newPlugin.Name = Name;
+            newPlugin.Computer = Computer;
+            newPlugin.TestQueue = TestQueue;
+
+            return newPlugin;
+        }
+
+        #endregion Inherited from IExecutorPlugin
 
-        private void RunTest(Test test)
+        private void RunTest(NUnitTest test)
         {
             // TODO: Run the test.
+            string baseOutputFile = StringUtils.FormatInvariant(@"{0}\{1}", Options.OutputDirectory, test.TestName);

```

Comment:
Should use the output directory of the Computer if it exists, otherwise use the one in Options.